.github/workflows/test.yml

```diff
@@ -26,10 +26,29 @@ jobs:
         with:
           python-version: "${{ matrix.python-version }}"
 
-      - name: "Install dependencies in pyproject.toml"
+      - name: "Install test and project dependencies"
         run: |
+          # Project dependencies from pyproject.toml
+          # NOTE: Also builds viscm. How do we avoid this?
           pip install .
-          pip install pytest pytest-cov ${{ matrix.pyqt-dependency }}
+
+          # Test dependencies
+          pip install pytest pytest-cov pytest-qt pytest-xvfb ${{ matrix.pyqt-dependency }}
+          # pytest-qt CI dependencies: https://pytest-qt.readthedocs.io/en/latest/troubleshooting.html#github-actions
+          sudo apt update
+          sudo apt install -y \
+            xvfb \
+            libxkbcommon-x11-0 libxcb-icccm4 libxcb-image0 libxcb-keysyms1 libxcb-randr0 libxcb-render-util0 libxcb-xinerama0 libxcb-xfixes0 x11-utils
 
       - name: "Run tests"
         run: "make test"
+        env:
+          # In Pythons >= 3.10, tests fail with `RuntimeError: Invalid DISPLAY
+          # variable`, unless this variable is set:
+          MPLBACKEND: "Agg"
+
+      - name: "Install mypy"
```

Comment:
Added to this workflow because we already have the project dependencies installed here. But we don't have a reason to run `mypy` on 4 different Python versions. Creating a Python environment is expensive, but so is running the type checker. I don't have strong feelings about this tradeoff :shrug: Happy to make this a separate workflow.